storage_service/common/utils.py

```diff
@@ -637,15 +661,33 @@ def generate_checksum(file_path, checksum_type="md5"):
     """
     Returns checksum object for `file_path` using `checksum_type`.
 
+    If `file_path` is a directory (i.e. an uncompressed package), return the
+    checksum of the Bag tag-manifest file. This allows us to later validate
+    that this is the correct AIP.
+
     If checksum_type is not a valid checksum, ValueError raised by hashlib.
     """
     checksum = hashlib.new(checksum_type)
+
+    if os.path.isdir(file_path):
+        file_path = find_tag_manifest(file_path)
+
     with open(file_path, "rb") as f:
         for chunk in iter(lambda: f.read(128 * checksum.block_size), b""):
             checksum.update(chunk)
     return checksum
 
 
+def find_tag_manifest(file_path):
+    """Return the path to a Bag's tagmanifest file."""
+    if not os.path.isdir(file_path):
+        return
+
+    for bag_file in os.listdir(file_path):
+        if bag_file.startswith("tagmanifest"):
```

Comment:
The spec says that [a bag may contain one or more tag manifest](https://www.rfc-editor.org/rfc/rfc8493.html#section-2.2.1), should we look for them following an order? (sha512, sha256, md5)?

Reply:
Also, they should always end in ```.txt```